src/callbacks_step/amr_dg2d.jl

```diff
@@ -1,3 +1,95 @@
+# Redistribute data for load balancing after partitioning the mesh
+function redistribute_data!(u_ode::AbstractVector, mesh::TreeMesh{2}, equations, 
+                            dg::DGSEM, cache, old_mpi_ranks_per_cell)
+  # Retain current solution data
```

Comment:
Please create an issue that one should probably short-circuit this routine if there is nothing to do. AFAICT, right now all data structures are recreated even if there is nothing to redistribute. That means, this operation needs to be performed *each* time the mesh has been changed, even if the current rank is not affected.

Reply:
Shouldn't a simple `if cache.elements.cell_ids == local_leaf_cells` do the job? If the elements' cell ids (before re-initializing the containers) are the same as the ones that are assigned to this rank by the mesh, there won't be any rebalancing to do for this rank.

Reply:
You could be right! If it's not too much to do, please check if it works and add it to this PR.